rider-fsharp/protocol/src/kotlin/model/RdFSharpTypeProvidersLoaderModel.kt

```diff
@@ -0,0 +1,157 @@
+package model
+
+import com.jetbrains.rider.model.nova.ide.SolutionModel
+import com.jetbrains.rd.generator.nova.*
+import com.jetbrains.rd.generator.nova.PredefinedType.*
+import com.jetbrains.rd.generator.nova.csharp.CSharp50Generator
+
+@Suppress("unused")
+object RdFSharpTypeProvidersLoaderModel : Root(CSharp50Generator(
+        FlowTransform.Symmetric,
+        "JetBrains.ReSharper.Features.WinForms.Designer.Protocol.Model")
+) {
+    private val Type = structdef {
+
+    }
+
+    private val ParameterInfo = structdef {
+
+    }
+
+    private val ResolutionEnvironment = structdef {
+        field("resolutionFolder", string)
+        field("outputFile", string) //string option
+        field("showResolutionMessages", bool)
+        field("referencedAssemblies", array(string))
+        field("temporaryFolder", string)
+    }
+
+    private val ProvidedNamespace = structdef {
+        call("getNestedNamespaces", void, immutableList(string))
+        call("getTypes", void, immutableList(Type))
+        call("resolveTypeName", string, Type)
+
+        field("namespaceName", string)
+    }
+/*
+    private val TypeProvider = structdef"TypeProvider") {
+        sink("invalidate", void).async
+
+        call("getNamespaces", void, immutableList(string))
+        call("applyStaticArguments", Type, immutableList(ParameterInfo))
+        call("getInvokerExpression", int, int)
+        call("getGeneratedAssemblyContents", void, immutableList(byte))
+    }*/
+
+    private val ApplyStaticArgumentsParameters = structdef {
+        field("typeProviderId", int)
+        field("typeWithoutArguments", string)
+        field("typePathWithArguments", array(string))
+        field("staticArguments", array(string))
+    }
+
+    private val GetStaticArgumentsParameters = structdef {
+        field("typeProviderId", int)
+        field("typeWithoutArguments", string)
+    }
+
+    private val GetInvokerExpressionParameters = structdef {
+        field("typeProviderId", int)
+        field("syntheticMethodBase", string)
+        field("parameters", string)
+    }
+
+    private val GetGeneratedAssemblyContentsParameters = structdef {
+        field("typeProviderId", int)
+        field("syntheticMethodBase", string)
+        field("parameters", string)
+    }
+
+    private val pos = structdef {

```

Comment:
Let's use `Rd` prefix for protocol types. It'll be something like `RdPos` here.

Reply:
> Let's use `Rd` prefix for protocol types. It'll be something like `RdPos` here.

fixed